doc/user-guide/windowing.md

```diff
@@ -0,0 +1,187 @@
+
+```text
+1, 5, 10, 15, 20, 25, 30, 35, 40
+|---------|                         [0  - 14]
+   |----------|                     [5  - 19]
+      |-----------|                 [10 - 24]
+          |-----------|             [15 - 29]
+              |-----------|         [20 - 34]
+                  |-----------|     [25 - 39]
+```
+
+### Units
+
+Onyx allows you to specify range and slide values in different magnitudes of units, so long as the units can be coverted to the same unit in the end. For example, you can specify the range in minutes, and the slide in seconds. Any value that requires units takes a vector of two elemenets. The first element represents the value, and the second the unit. For example, window specifications denoting range and slide might look like:
+
+```clojure
+{:window/range [1 :minute]
+ :window/slide [30 :seconds]}
+```
+
+See the information model for all supported units. You can use a singular form (e.g. `:minute`) instead of the plural (e.g. `:minutes`) where it makes sense for readability.
+
+Onyx is also capable of sliding by `:elements`. This is often referred to as "slide-by-tuple" in research. Onyx doesn't require a time-based range and slide value. Any totally ordered value will work equivalently.
+
+### Aggregation
+
+Windows allow you accrete data over time. Sometimes, you want to store all the data. Othertimes you want to incrementally compact the data. Window specifications must provide a `:window/aggregation` key. We'll go over an example of every type of aggregation that Onyx supports.
+
+#### `:conj`
+
+The `:conj` aggregation is the simplest. It collects segments for this window and retains them in a vector, unchanged.
+
+```clojure
+{:window/id :collect-segments
+ :window/task :identity
+ :window/type :sliding
+ :window/aggregation :conj
+ :window/window-key :event-time
+ :window/range [30 :minutes]
+ :window/slide [5 :minutes]
+ :window/doc "Collects segments on a 30 minute window sliding every 5 minutes"}
+```
+
+#### `:count`
+
+The `:count` operation counts the number of segments in the window.
+
+```clojure
+{:window/id :count-segments
+ :window/task :identity
+ :window/type :fixed
+ :window/aggregation :count
+ :window/window-key :event-time
+ :window/range [1 :hour]
+ :window/doc "Counts segments in one hour fixed windows"}
+```
+
+#### `:sum`
+
+The `:sum` operation adds the values of `:window/sum-key` for all segments in the window.
+
+```clojure
+{:window/id :sum-ages
+ :window/task :identity
+ :window/type :fixed
+ :window/aggregation :sum
+ :window/sum-key :age
+ :window/window-key :event-time
```

Comment:
As an alternative to using :window/aggregation X, :window/X-key, how about:
`:window/aggregation [aggregation key]`, e.g. `:window/aggregation [:sum :age]`. This makes it a lot easier to switch aggregation types programatically.